File: solutions/3/3.py
def largestPrimeFactor(number):
    divisor = 2

    while True:
        while number % divisor == 0:
            number = number / divisor

        if number == 1:
            break

        divisor += 1

    return divisor

def main():
    result = largestPrimeFactor(600851475143)

    print(result)

if __name__ == '__main__':
    main()

File: solutions/3/3.js
function largestPrimeFactor(number) {
    let divisor = 2;

    while (true) {
        while (number % divisor == 0) {
            number = number / divisor;
        }

        if (number == 1) {
            break;
        }

        divisor += 1;
    }

    return divisor;
}

function main() {
    let result = largestPrimeFactor(600851475143);

    console.log(result);
}

main();
